test/src/test_beam_likelihood.cpp

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (c) 2020, the mcl_3dl authors
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of the copyright holder nor the names of its
+ *       contributors may be used to endorse or promote products derived from
+ *       this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <cmath>
+#include <cstddef>
+#include <vector>
+
+#include <gtest/gtest.h>
+
+#include <ros/ros.h>
+
+#include <mcl_3dl/chunked_kdtree.h>
+#include <mcl_3dl/lidar_measurement_models/lidar_measurement_model_beam.h>
+#include <mcl_3dl/vec3.h>
+
+TEST(BeamModel, LikelihoodFunc)
+{
+  pcl::PointCloud<mcl_3dl::LidarMeasurementModelBase::PointType> pc;
+  for (float y = -0.2; y <= 0.2; y += 0.1)
+  {
+    for (float z = -0.2; z <= 0.2; z += 0.1)
+    {
+      mcl_3dl::LidarMeasurementModelBase::PointType p;
+      p.x = 2;
+      p.y = y;
+      p.z = z;
+      pc.push_back(p);
+    }
+  }
+  pcl::PointCloud<mcl_3dl::LidarMeasurementModelBase::PointType> pc_map = pc;
+  {
+    // Add dummy points to expand DDA grid size.
+    mcl_3dl::LidarMeasurementModelBase::PointType p;
+    p.x = -100;
+    p.y = 100;
+    p.z = 0;
+    pc_map.push_back(p);
+    p.x = 100;
+    p.y = -100;
+    p.z = 4;
```

Comment:
```suggestion
    p.x = -100.05;
    p.y = 100;
    p.z = -0.05;
    pc_map.push_back(p);
    p.x = 100;
    p.y = -100.05;
    p.z = 4;
```
Please avoid that points exists on the edges of voxels. By this change, each point is moved to the centers of voxel, and the result becomes stable.